consensus/core/src/metrics.rs

```diff
@@ -98,15 +99,25 @@ enum MetricType {
 impl Metrics {
     pub(crate) fn update_scoring_metrics_on_eviction(
         &self,
-        validator: AuthorityIndex,
+        authority_index: AuthorityIndex,
         hostname: &str,
+        recent_refs: &BTreeSet<BlockRef>,
         eviction_round: u32,
-        last_evicted_round: u32,
-        recent_refs_by_authority: &BTreeSet<BlockRef>,
+        last_eviction_round: u32,
         threshold_clock_round: u32,
     ) -> Option<StoredScoringMetricsU64> {
+        // threshold_clock_round will be always at least 1. However, this method will
+        // panic if threshold_clock_round = 0 (subtraction with overflow). We
```

Comment:
To prevent accidental panics I think we could additionally use `saturating_sub` function.

Reply:
done